### PR TITLE
#52 refactor tasks

### PR DESCRIPTION
**Pull request by Rot127**
This PR ticks most boxes in https://github.com/rizinorg/rz-hexagon/issues/52.

Namely:

- [x] ~~Fix grammar/spelling issues (`handwritten` -> `hand-written`)~~
- [x] Move the foloowing pattern in its own function
  ```python
  with open("handwritten/hexagon_il_h/include.h") as include:
    set_pos_after_license(include)
    code += "".join(include.readlines())
  ```
- [x] Move the following pattern in its own function:
  ```python
  if compare_src_to_old_src(code, path):
     self.unchanged_files.append(path)
     return
  with open(path, "w+") as dest:
     dest.writelines(code)
     log("analysis_hexagon.c written to {}".format(path), LogLevel.INFO)
   ```
- [x] Replaced `"..".format()` patterns with `f"..."` formatting in https://github.com/rizinorg/rz-hexagon/pull/60 
- [x] Refactor `parse_instruction()`
  The method for parsing a Duplex or normal instruction are almost the same.
  The method should be in `InstructionTemplate` from which `DuplexInstruction` and `Instruction` inherit.
  
  Methods in question:
  https://github.com/rizinorg/rz-hexagon/blob/7df09f53826546b8e0a328021813882e13e21de1/DuplexInstruction.py#L99
  https://github.com/rizinorg/rz-hexagon/blob/7df09f53826546b8e0a328021813882e13e21de1/Instruction.py#L106

- [x] Update license date
- [x] Remove or include out-commented code (mostly logs)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ This plugin is under continuous work. So checkout the Github issues for missing
 - For formatting we need `clang-format-13`. If it is not available on your distribution, you can install it from https://apt.llvm.org/.
 
 - Python requirements are in `requirements.txt`
+- As a developer you also need `black, flake8, reuse`.
```
Comment (XVilka) [inline README.md:24]: `black`, `flake8`, `reuse`

File: README.md [hunk 2]
```diff
@@ -88,6 +90,9 @@ So here are some good to know points for porting:
 
 # Development info
 
+**Before you open a PR please run: `black -l 120 *.py && flake8 --max-line-length=120 && reuse lint`** and fix the warnings.
```
Comment (XVilka) [inline README.md:93]: Are those in CI as well?

**Review by XVilka — APPROVED** [review]

**Review by thestr4ng3r — CHANGES_REQUESTED** [review]
"handwritten" is the correct form: https://dictionary.cambridge.org/de/worterbuch/englisch/handwritten

File: README.md (hunk 2 above)
Comment (Rot127) [reply README.md:93]: Not yet.

**Review by thestr4ng3r — APPROVED** [review]

**Pull request merged**